app/db/interface.py

```diff
@@ -110,15 +136,49 @@ def _query_payments(self, payment_attributes):
         :return: <list[Payment]> a list of the Payment items returned by the query
         """
         try:
+
             payment_query = self.db.session.query(Payment).filter_by(**payment_attributes)
             payments = payment_query.all()
             return payments
 
         except exc.SQLAlchemyError:
             raise PaymentServiceQueryError('Could not retrieve payment items due to query error with given attributes')
 
+# UTILITY FUNCTIONS
+    def is_valid_put(self,old_data,new_data):
+        valid = False
+        valid_detail = False
+        try:
+            if new_data['is_removed'] and new_data['is_removed'] == True :
+                 raise DataValidationError('Invalid payment: Changes to is_removed field not allowed')
```

Comment:
why check two conditions here?
`if new_data['is_removed'] and new_data['is_removed'] == True :`

don't we just want to confirm whether or not `old_data` has been removed ?